.travis.yml

```diff
@@ -0,0 +1,44 @@
+language: c
+dist: trusty   # Travis build machine is Ubuntu Trusty 14.04 x64
+sudo: required # A sudo enabled, full VM is used for building instead of a docker container
+
+git:
+  submodules: false
+
+before_install:
+  - sudo apt-get update -qq
+  # Pre-reqs for building omsagent
+  - sudo apt-get install -qq pkg-config make ruby bison g++ rpm librpm-dev libpam0g-dev libssl-dev libmysqlclient-dev
+  - rvm use system                 # Use system ruby instead of ruby provided by rvm
+  - sed -i 's/--no-ri//' ~/.gemrc  # ri data from installed gems is required for our build
+
+# Travis-CI automatically clones the OMS-Agent-for-Linux repo since this is where this .travis.yml file is located
+# We will ignore this clone and clone the superproject Build-OMS-Agent-for-Linux as follows
+
+install:
+  - cd ../..
+  - git clone https://github.com/sugr4/Build-OMS-Agent-for-Linux.git sugr4/Build-OMS-Agent-for-Linux
+  - cd sugr4/Build-OMS-Agent-for-Linux
+  # change url from ssh to https for submodules, travis-ci cannot download a submodule that isn’t on a public url
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules 
+  - git submodule update --init --recursive dsc omi opsmgr opsmgr-kits pal
+  - git clone https://github.com/sugr4/OMS-Agent-for-Linux.git omsagent 
+  - cd omsagent
+  # change url from ssh to https for submodules ruby and fluentd
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules 
+  - git submodule update --init --recursive
+
+before_script:
+  - cd ..
+  - git checkout master && git submodule foreach git checkout master
+  - cd omsagent
+  # PR validation during check-in will be setup in github in the settings section of our repo
+  - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then git fetch origin refs/pull/$TRAVIS_PULL_REQUEST/head ; git checkout FETCH_HEAD; fi 
+  - cd build
+  # pwd: /home/travis/build/sugr4/Build-OMS-Agent-for-Linux/omsagent/build
+  # Since 'build' is a keyword for travis-CI, change Makefile to substitute 'omsagent/build' instead of 'build'
+  - sed -i 's/(subst \/build,,$(CURDIR))/(subst omsagent\/build,omsagent,$(CURDIR))/' Makefile 
+
+script:
+  - sudo ./configure --enable-debug
+  - sudo make && sudo make unittest # System tests are not run because OMS-Agent-for-Linux-testconfig.git is a private repo
```

Comment:
Should we have to run these commands with "sudo"? I thought we had resolved that issue.